File: 366. 斐波纳契数列.cpp
class Solution {
public:
    /**
     * @param n: an integer
     * @return: an ineger f(n)
     */
    int fibonacci(int n) {
        if (n == 1) return 0;
        if (n == 2) return 1;
        int prev = 0;
        int curr = 1;
        for (int i = 3; i <= n; ++i) {
            int temp = curr;
            curr += prev;
            prev = temp;
        }
        return curr;
    }
};

// 总耗时 288 ms
// 您的提交打败了 37.20% 的提交!

File: Python3/366. 斐波纳契数列.py
class Solution:
    """
    @param n: an integer
    @return: an ineger f(n)
    """
    def fibonacci(self, n):
        if n == 1 :
            return 0
        if n == 2 :
            return 1
        prev = 0
        curr = 1
        for i in range(3, n + 1) :
            prev, curr = curr, curr + prev
        return curr

# 总耗时 793 ms
# 您的提交打败了 95.20% 的提交!